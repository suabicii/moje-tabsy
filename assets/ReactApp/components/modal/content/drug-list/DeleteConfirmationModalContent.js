import React from "react";
import {useSelector} from "react-redux";

function DeleteConfirmationModalContent({setIsModalOpen, confirmDeletion, drugId}) {
    const darkMode = useSelector(state => state.darkMode);

    return (
        <div className="container text-center" role="dialog" data-testid="delete-confirmation">
            <h2 className="text-danger-emphasis">Czy na pewno chcesz usunąć dany lek/suplement?</h2>
            <div className="grid mt-4">
                <button className="btn btn-danger me-lg-2" data-testid="yes" onClick={() => {
                    confirmDeletion(drugId);
                    setIsModalOpen(false);
                }}>Tak <i className="fa-solid fa-check"></i>
                </button>
                <button className={`btn btn-${darkMode ? 'light' : 'dark'} ms-lg-2 mt-lg-0 mt-2`} data-testid="no" onClick={() => {
                    setIsModalOpen(false);
                }}>Nie <i className="fa-solid fa-xmark"></i>
                </button>
            </div>
        </div>
    );
}

export default DeleteConfirmationModalContent;