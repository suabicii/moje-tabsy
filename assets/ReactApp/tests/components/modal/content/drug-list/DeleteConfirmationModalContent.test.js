/**
 * @jest-environment jsdom
 * */
import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import DeleteConfirmationModalContent
    from "../../../../../components/modal/content/drug-list/DeleteConfirmationModalContent";
import {fireEvent, render, screen} from "@testing-library/react";

it('should correctly render DeleteConfirmationModalContent', () => {
    const renderer = new ReactShallowRenderer();
    const setIsModalOpen = jest.fn();

    renderer.render(<DeleteConfirmationModalContent setIsModalOpen={setIsModalOpen}/>);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});

it('should close modal after clicking "No"', () => {
    const setIsModalOpen = jest.fn();
    render(<DeleteConfirmationModalContent setIsModalOpen={setIsModalOpen}/>);

    fireEvent.click(screen.getByTestId('no'));

    expect(setIsModalOpen).toBeCalledWith(false);
});