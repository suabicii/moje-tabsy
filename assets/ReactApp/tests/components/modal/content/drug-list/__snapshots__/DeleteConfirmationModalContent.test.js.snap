// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render DeleteConfirmationModalContent 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <DeleteConfirmationModalContent
    confirmDeletion={[MockFunction]}
    drugId={1}
    setIsModalOpen={[MockFunction]}
  />
</Provider>
`;
